tests/test_shuffle.py

```diff
@@ -1,12 +1,70 @@
+import uuid
+
 import dask
+import dask.dataframe as dd
+import pytest
+
+
+@pytest.fixture(scope="session")
+def s3_stability_url(s3, s3_bucket):
+    # Ensure that the stability-scratch directory exists,
+    # but do NOT reomove it as multiple test runs could be
+    # accessing it at the same time
+    stability_url = f"{s3_bucket}/stability-scratch"
+    s3.mkdirs(stability_url, exist_ok=True)
+    return stability_url
+
+
+@pytest.fixture(scope="session")
+def s3_test_data_url(s3, s3_stability_url):
+    # Unique, because multiple tests are accessing the bucket
+    test_url = f"{s3_stability_url}/test-data-{uuid.uuid4().hex}"
+
+    try:
+        s3.makedirs(test_url)
+        yield test_url
+    finally:
+        s3.rm(test_url, recursive=True)
+
 
+@pytest.fixture
+def s3_write_url(s3, s3_stability_url):
+    # Unique, because multiple tests are accessing the bucket
+    write_url = f"{s3_stability_url}/test-output-{uuid.uuid4().hex}"
 
-def test_shuffle_simple(small_client):
+    try:
+        s3.makedirs(write_url)
+        yield write_url
+    finally:
+        s3.rm(write_url, recursive=True)
+
+
+@pytest.fixture
+def shuffle_dataset(small_client, s3_test_data_url, s3_storage_options):
+    """Produces a ~80GB dataset which is about the memory limit of the cluster"""
     df = dask.datasets.timeseries(
-        start="2000-01-01", end="2000-12-31", freq="1s", partition_freq="1D"
+        start="2000-01-01", end="2000-12-31", freq="50ms", partition_freq="1D"
     )
 
-    sdf = df.shuffle(on="x")
+    df.to_parquet(
+        f"s3://{s3_test_data_url}",
+        compute=True,
+        overwrite=True,
+        storage_options=s3_storage_options,
+    )
```

Comment:
Just curious, how long does this write take?

Reply:
The whole test takes less than a minute, so it seems negligble at present -- so inlining seems sensible.
```bash
============================= test session starts ==============================
platform linux -- Python 3.7.12, pytest-7.1.1, pluggy-1.0.0 -- /usr/share/miniconda3/envs/test/bin/python
cachedir: .pytest_cache
rootdir: /home/runner/work/coiled-runtime/coiled-runtime, configfile: setup.cfg
plugins: anyio-3.5.0
collecting ... collected 1 item

tests/test_shuffle.py::test_shuffle_simple PASSED

=============================== warnings summary ===============================
../../../../../usr/share/miniconda3/envs/test/lib/python3.7/site-packages/coiled/_beta/cluster.py:11
  /usr/share/miniconda3/envs/test/lib/python3.7/site-packages/coiled/_beta/cluster.py:11: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
    from collections import Iterable

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================== slowest durations ===============================
3060.41s call     tests/test_shuffle.py::test_shuffle_simple
109.42s setup    tests/test_shuffle.py::test_shuffle_simple
0.35s teardown tests/test_shuffle.py::test_shuffle_simple
================== 1 passed, 1 warning in 3170.43s (0:52:50) ===================
```

Reply:
@sjperkins I think you might have missed the comment above, do you have an estimate?

Reply:
Apologies, I was holding back because of the run time discrepancies James and I were seeing with `compute={True,False}` on `to_parquet`.

When the test is working as expected it takes ~5m30s. I estimate from watching the cluster that it takes roughly 30s to write the testing dataset.